### PR TITLE
[error-issue] Add helper class to build error report issues

### PR DESCRIPTION
**Pull request by rcebulko**
Part of ampproject/amphtml#27675

Probably easiest to just review each file, rather than commit-by-commit.
Top-down view:
```
 PASS  test/issue_builder.test.ts
  IssueBuilder
    title
      ✓ contains the error message (2ms)
    labels
      ✓ contains the error report label
    bodyDetails
      ✓ links to the error (1ms)
      ✓ records the date first seen (1ms)
      ✓ records the daily frequency
    bodyStacktrace
      ✓ renders the indented stacktrace in markdown (1ms)
    bodyNotes
      ✓ includes blame info for each line of the stacktrace
      ✓ suggests possible assignees, if known (1ms)
    body
      ✓ generates the full issue body markdown (1ms)

 PASS  test/utils.test.ts
  formatDate
    ✓ produces a short date format (2ms)

------------------|---------|----------|---------|---------|-------------------
File              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------|---------|----------|---------|---------|-------------------
All files         |     100 |      100 |     100 |     100 |                   
 issue_builder.ts |     100 |      100 |     100 |     100 |                   
 utils.ts         |     100 |      100 |     100 |     100 |                   
------------------|---------|----------|---------|---------|-------------------
```

### CODE REVIEW

**Comment by rcebulko** [comment]
Friendly ping :upside_down_face:

File: error-issue/src/issue_builder.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  get labels(): Array<string> {
+    return ['Type: Error Report'];
+  }
+
+  get bodyDetails(): string {
+    return [
+      'Details',
+      '---',
+      `**Error report:** [link](go/ampe/${this.errorId})`,
```
Comment (danielrozenberg) [inline error-issue/src/issue_builder.ts:50]: Can you make this a public GCloud link? If somebody doesn't have permission on this they'll get a 401/403, and non-googlers with permissions won't have to jump through hoops to get to the error report :)

File: error-issue/src/issue_builder.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    prNumber,
+  }: BlameRange): string {
+    return (
+      `\`@${author}\` modified \`${path}:${startingLine}-${endingLine}\`` +
+      ` in #${prNumber} (${formatDate(committedDate)})`
+    );
+  }
+
+  possibleAssignees(): Array<string> {
+    return [];
```
Comment (danielrozenberg) [inline error-issue/src/issue_builder.ts:83]: `TODO`s are cheap :D

File: error-issue/test/issue_builder.test.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {IssueBuilder} from '../src/issue_builder';
+import {ErrorReport} from '../src/types';
+
+const FULL_ISSUE_BODY = `Details
```
Comment (danielrozenberg) [inline error-issue/test/issue_builder.test.ts:20]: This is only used once, how about inlining it? I actually prefer that you put this in a .txt file and read that file in the test - it's pretty long

File: error-issue/src/issue_builder.ts (hunk 1 above)
Comment (rcebulko) [reply error-issue/src/issue_builder.ts:50]: Punting on this, as I haven't found any way to make accessible a Pantheon project created internally.

File: error-issue/src/issue_builder.ts (hunk 2 above)
Comment (rcebulko) [reply error-issue/src/issue_builder.ts:83]: Done.

File: error-issue/test/issue_builder.test.ts (hunk 3 above)
Comment (rcebulko) [reply error-issue/test/issue_builder.test.ts]: Done.

**Comment by estherkim** [comment]
I'm confused. Is this already deployed, but not committed to amp-github-apps?

**Comment by rcebulko** [comment]
> I'm confused. Is this already deployed, but not committed to amp-github-apps?

There is a prototype of this that I've been using to validate the design/interactions with APIs. So, somewhat? It's not actively monitoring anything, but right now I use a bookmarklet to trigger and auto-file issues

**Comment by estherkim** [comment]
Hmmm.. the whole point of automating deployments from master was to avoid situations like this.

I can see that issues are already being created on amphtml using ampprojectbot's credentials, from code on your local workspace that hasn't been reviewed by any of us. I know it seems benign, but shouldn't bots we deploy go through the proper review process before they are deployed?

**Comment by rcebulko** [comment]
It's not a deployed bot, it's a bookmarklet I've been using to manually create issues while I monitor errors.

**Comment by estherkim** [comment]
Oh I see

**Pull request merged**